### PR TITLE
feat(route): 增加了六个美国博物馆的展讯

### PR DESCRIPTION
**Pull request by chazeon**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/brooklynmuseum/exhibitions
/brooklynmuseum/exhibitions/current
/brooklynmuseum/exhibitions/upcoming
/brooklynmuseum/exhibitions/past
/metmuseum/exhibitions
/metmuseum/exhibitions/current
/metmuseum/exhibitions/upcoming
/metmuseum/exhibitions/past
/guggenheim/exhibitions
/jewishmuseum/exhibitions
/mcachicago/exhibitions
/newmuseum/exhibitions
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [x] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

增加了六个美国博物馆的展讯
- 部分 `pubDate` 设置为了展览开始日
- 部分全文获取有效

### CODE REVIEW

File: lib/v2/brooklynmuseum/exhibitions.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            link = `https://www.brooklynmuseum.org/exhibitions/`;
+            break;
+        default:
+            link = `https://www.brooklynmuseum.org/exhibitions/${state}`;
+    }
+
+    ctx.state.data = await buildData({
+        link,
+        url: link,
+        title: `Brooklyn Museum - Exhibitions`,
```
Comment (TonyRL) [inline lib/v2/brooklynmuseum/exhibitions.js:19]:
```suggestion
        title: 'Brooklyn Museum - Exhibitions',
```

File: lib/v2/jewishmuseum/exhibitions.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const buildData = require('@/utils/common-config');
+
+module.exports = async (ctx) => {
+    const link = `https://thejewishmuseum.org/exhibitions`;
```
Comment (TonyRL) [inline lib/v2/jewishmuseum/exhibitions.js:4]:
```suggestion
    const link = 'https://thejewishmuseum.org/exhibitions';
```

File: lib/v2/jewishmuseum/exhibitions.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+    ctx.state.data = await buildData({
+        link,
+        url: link,
+        title: `Jewish Museums - Exhibitions`,
```
Comment (TonyRL) [inline lib/v2/jewishmuseum/exhibitions.js:9]:
```suggestion
        title: 'Jewish Museums - Exhibitions',
```

File: lib/v2/jewishmuseum/exhibitions.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        item: {
+            item: 'article',
+            title: `$('article.exhibition h3').text()`,
+            link: `$('article.exhibition > a').attr('href')`,
+        },
+    });
+};
```
Comment (TonyRL) [inline lib/v2/jewishmuseum/exhibitions.js:16]: There's one empty item in `/jewishmuseum/exhibitions`

<details>
<summary>Screenshot</summary>

image
</details>

File: lib/v2/jewishmuseum/radar.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+    'thejewishmuseum.org': {
+        _name: 'Jewish Museum',
+        '.': [
+            {
+                title: 'Exhibitions',
+                docs: 'http://docs.rsshub.app/en/travel.html#the-jewish-museum',
```
Comment (TonyRL) [inline lib/v2/jewishmuseum/radar.js:7]:
```suggestion
                docs: 'https://docs.rsshub.app/en/travel.html#the-jewish-museum',
```

File: lib/v2/metmuseum/exhibitions.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+module.exports = async (ctx) => {
+    const searchType = ctx.params.state ?? 'current';
+
+    const url = `https://www.metmuseum.org/ghidorah/ExhibitionListing/Search?searchType=${searchType}`;
+
+    const response = await got({
+        url,
+        method: 'GET',
+        insecureHTTPParser: true,
```
Comment (TonyRL) [inline lib/v2/metmuseum/exhibitions.js:21]:
```suggestion
```
https://github.com/sindresorhus/got/issues/1794#issuecomment-886050367

File: lib/v2/mcachicago/exhibitions.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+const buildData = require('@/utils/common-config');
+
+module.exports = async (ctx) => {
+    const link = `https://mcachicago.org/exhibitions`;
+
+    ctx.state.data = await buildData({
+        link,
+        url: link,
+        title: `MCA Chicago - Exhibitions`,
```
Comment (TonyRL) [inline lib/v2/mcachicago/exhibitions.js:9]:
```suggestion
        title: 'MCA Chicago - Exhibitions',
```

File: lib/v2/mcachicago/exhibitions.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+const buildData = require('@/utils/common-config');
+
+module.exports = async (ctx) => {
+    const link = `https://mcachicago.org/exhibitions`;
```
Comment (TonyRL) [inline lib/v2/mcachicago/exhibitions.js:4]:
```suggestion
    const link = 'https://mcachicago.org/exhibitions';
```

File: lib/v2/metmuseum/exhibitions.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
+const got = require('@/utils/got');
+
+function generateExhibitionItem(result) {
+    return {
+        title: result.title,
+        link: `https://www.metmuseum.org${result.url}`,
+        description: result.description,
+        pubDate: result.startDate,
```
Comment (TonyRL) [inline lib/v2/metmuseum/exhibitions.js:8]:
```suggestion
        pubDate: parseDate(result.startDate),
```
docs: https://docs.rsshub.app/en/joinus/pub-date.html

File: lib/v2/newmuseum/exhibitions.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
+const buildData = require('@/utils/common-config');
+
+module.exports = async (ctx) => {
+    let link;
+    const state = ctx.query.state;
+
+    switch (state) {
+        case undefined:
+        case 'current':
+            link = `https://www.newmuseum.org/exhibitions/`;
```
Comment (TonyRL) [inline lib/v2/newmuseum/exhibitions.js:10]:
```suggestion
            link = 'https://www.newmuseum.org/exhibitions/';
```

File: lib/v2/newmuseum/exhibitions.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+            break;
+        default:
+            link = `https://www.newmuseum.org/exhibitions/${state}`;
+    }
+
+    ctx.state.data = await buildData({
+        link,
+        url: link,
+        title: `New Museum - Exhibitions`,
```
Comment (TonyRL) [inline lib/v2/newmuseum/exhibitions.js:19]:
```suggestion
        title: 'New Museum - Exhibitions',
```

File: lib/v2/guggenheim/exhibitions.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const vm = require('vm');
+
+function parseDate(dateObj) {
+    return Date.parse(`${dateObj.month} ${dateObj.day}, ${dateObj.year}`);
+}
```
Comment (TonyRL) [inline lib/v2/guggenheim/exhibitions.js:7]:
```suggestion
const { parseDate} = require('@/utils/parse-date');
```
docs: https://docs-rsshub.pages.dev/en/joinus/pub-date.html

File: lib/v2/guggenheim/exhibitions.js [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+
+    const response = await got({
+        url: link,
+        method: 'GET',
+    });
+
+    const code = cheerio.load(response.data)('#app-js-extra').html();
+    const context = vm.createContext();
+    vm.runInContext(code, context);
+    const data = context.bootstrap;
```
Comment (TonyRL) [inline lib/v2/guggenheim/exhibitions.js:20]:
```suggestion
    const data = JSON.parse(cheerio.load(response.data)('#app-js-extra').html().match(/var bootstrap = (.*);\svar navFooter/)[1]);
```
no need to use `vm`

File: lib/v2/guggenheim/exhibitions.js [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+
+    ctx.state.data = {
+        link,
+        url: link,
+        title: `The Guggenheim Museums and Foundation - Exhibitions`,
+        item: items.map((ex) => ({
+            title: ex.title,
+            link: `https://www.guggenheim.org/exhibition/${ex.slug}`,
+            description: ex.excerpt,
+            pubDate: ex.dates ? parseDate(ex.dates.start) : null,
```
Comment (TonyRL) [inline lib/v2/guggenheim/exhibitions.js:32]:
```suggestion
            pubDate: ex.dates ? parseDate(`${ex.dates.start.month} ${ex.dates.start.day}, ${ex.dates.start.year}`, 'MMMM D, YYYY') : null,
```

File: lib/v2/guggenheim/exhibitions.js [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+    const context = vm.createContext();
+    vm.runInContext(code, context);
+    const data = context.bootstrap;
+    const exhibitions = data.initial.main.posts.featuredExhibitions;
+    const items = [].concat(exhibitions.past.items ?? [], exhibitions.on_view.items ?? [], exhibitions.upcoming.items ?? []);
+
+    ctx.state.data = {
+        link,
+        url: link,
+        title: `The Guggenheim Museums and Foundation - Exhibitions`,
```
Comment (TonyRL) [inline lib/v2/guggenheim/exhibitions.js:27]:
```suggestion
        title: 'The Guggenheim Museums and Foundation - Exhibitions',
```

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/guggenheim/exhibitions.js [hunk 16]
```diff
@@ -0,0 +1,35 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const vm = require('vm');
+
+function parseDate(dateObj) {
+    return Date.parse(`${dateObj.month} ${dateObj.day}, ${dateObj.year}`);
+}
+
+module.exports = async (ctx) => {
+    const link = `https://www.guggenheim.org/exhibitions`;
```
Comment (TonyRL) [inline lib/v2/guggenheim/exhibitions.js:10]:
```suggestion
    const link = 'https://www.guggenheim.org/exhibitions';
```

**Comment by chazeon** [comment]
@TonyRL **About the `insecureHTTPParser`**: I used to get the `HPE_INVALID_HEADER_TOKEN` error, and adding this option indeed fix the error. Now even after removing this option this error no longer appears. Do you have any idea why I got `HPE_INVALID_HEADER_TOKEN` earlier?

**Comment by TonyRL** [comment]
> @TonyRL **About the `insecureHTTPParser`**: Yesterday I used to get the `HPE_INVALID_HEADER_TOKEN` error, and adding this `insecureHTTPParser` option indeed fix the error. Today even after removing this option this error no longer appears. Do you have any idea why I got `HPE_INVALID_HEADER_TOKEN` earlier?

There are some replies that reported it's an issue with Incapsula CDN in https://github.com/nodejs/node/issues/27711#issuecomment-553122893 and above, though one below said it's not just Incapsula.

Coincident?
```shell
$ nslookup www.metmuseum.org
Server:         192.168.1.250
Address:        192.168.1.250#53

Non-authoritative answer:
www.metmuseum.org       canonical name = bhpijji.x.incapdns.net.
Name:   bhpijji.x.incapdns.net
Address: 45.60.83.20
```

**Comment by chazeon** [comment]
> > @TonyRL **About the `insecureHTTPParser`**: Yesterday I used to get the `HPE_INVALID_HEADER_TOKEN` error, and adding this `insecureHTTPParser` option indeed fix the error. Today even after removing this option this error no longer appears. Do you have any idea why I got `HPE_INVALID_HEADER_TOKEN` earlier?
> 
> There are some replies that reported it's an issue with Incapsula CDN in [nodejs/node#27711 (comment)](https://github.com/nodejs/node/issues/27711#issuecomment-553122893) and above, though one below said it's not just Incapsula.
> 
> Coincident?
> 
> ```shell
> $ nslookup www.metmuseum.org
> Server:         192.168.1.250
> Address:        192.168.1.250#53
> 
> Non-authoritative answer:
> www.metmuseum.org       canonical name = bhpijji.x.incapdns.net.
> Name:   bhpijji.x.incapdns.net
> Address: 45.60.83.20
> ```

Good to know. This is likely to be the cause.

**Comment by chazeon** [comment]
I think all the issues have been resolved. Please take another look.

File: lib/v2/jewishmuseum/exhibitions.js [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+
+module.exports = async (ctx) => {
+    const link = 'https://thejewishmuseum.org/exhibitions';
+
+    ctx.state.data = await buildData({
+        link,
+        url: link,
+        title: 'Jewish Museums - Exhibitions',
+        item: {
+            item: '#current article.exhibition, #upcoming article.exhibition, #past article.exhibition',
```
Comment (TonyRL) [inline lib/v2/jewishmuseum/exhibitions.js:11]:
```suggestion
            item: '#current article.exhibition, #upcoming article, #past article.exhibition',
```
upcoming exhibition isn't shown

File: lib/v2/guggenheim/exhibitions.js [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+
+    ctx.state.data = {
+        link,
+        url: link,
+        title: 'The Guggenheim Museums and Foundation - Exhibitions',
+        item: items.map((ex) => ({
+            title: ex.title,
+            link: `https://www.guggenheim.org/exhibition/${ex.slug}`,
+            description: ex.excerpt,
+            pubDate: ex.dates ? parseDate(ex.dates.start, 'MMMM D, YYYY') : null,
```
Comment (TonyRL) [inline lib/v2/guggenheim/exhibitions.js:26]: Built-in `parseDate()` accept `Date` object or a date string but not a custom JSON
```suggestion
            pubDate: ex.dates ? parseDate(`${ex.dates.start.month} ${ex.dates.start.day}, ${ex.dates.start.year}`, 'MMMM D, YYYY') : null,
```

File: docs/en/travel.md [hunk 19]
```diff
@@ -31,3 +39,27 @@ This route returns a list of flight deals (in most cases, 6 flight deals) for a
...
+## The Jewish Museum
+
+<RouteEn author="chazeon" example="/jewishmuseum/exhibitions" path="/jewishmuseum/exhibitions" />
+
+## The Metropolitan Museum of Art
+
+<RouteEn author="chazeon"
+    example="/metmuseum/exhibitions"
+    path="/metmusem/exhibitions/:state?"
+    :paramsDesc="['state of the exhibition: `current`，`past`, or `upcoming`, the default value is `current`']"
```
Comment (TonyRL) [inline docs/en/travel.md:64]:
```suggestion
    :paramsDesc="['state of the exhibition: `current`，`past`, or `upcoming`, the default value is `current`']" anticrawler="1"
```

File: docs/travel.md [hunk 20]
```diff
@@ -114,6 +114,38 @@ IATA 国际航空运输协会机场代码，参见[维基百科 国际航空运
...
+ path="/brooklynmuseum/exhibitions/:state?"
+ :paramsDesc="['展览进行的状态：`current` 对应展览当前正在进行，`past` 对应过去的展览，`upcoming` 对应即将举办的展览，默认为 `current`']"
+/>
+
+## 纽约大都会美术馆
+
+<Route author="chazeon"
+ example="/metmuseum/exhibitions"
+ path="/metmusem/exhibitions/:state?"
+ :paramsDesc="['展览进行的状态：`current` 对应展览当前正在进行，`past` 对应过去的展览，`upcoming` 对应即将举办的展览，默认为 `current`']"
```
Comment (TonyRL) [inline docs/travel.md:130]:
```suggestion
 :paramsDesc="['展览进行的状态：`current` 对应展览当前正在进行，`past` 对应过去的展览，`upcoming` 对应即将举办的展览，默认为 `current`']" anticrawler="1"
```

**Review by TonyRL — COMMENTED** [review]
It seems their Incapsula WAF has already blocked me

**Pull request merged**